### PR TITLE
Services Redesign

### PR DESCRIPTION
**Pull request by easternbloc**
* New accumulated KPIs
* uses new transactions data endpoint
* data presented in UI as a sortable table instead of the previous, alphabetized list
* table collapses to a list at narrow viewports (< iPad portrait)
* accessibility improvements include screenreader announcements of sort order changes

### CODE REVIEW

File: app/common/collections/dashboards.js [hunk 1]
```diff
@@ -55,6 +49,56 @@ function (Collection) {
...
+            };
+          }
+        });
+      });
+
+      _.each(axes, function (axis) {
+        var key = (axis.format === 'percent') ? 'percentages' : 'totals',
+          weightedAverage;
+
+        if (aggregatedValues[key] && aggregatedValues[key][axis.key] && aggregatedValues.totals.number_of_transactions.value) {
```
Comment (jcbashdown) [inline app/common/collections/dashboards.js]: Delete code that is redone in services and used by nothing else that extends dashboard

File: app/common/collections/services.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          kpi = _.findWhere(aggregatedValues[key], {key: axis.key});
+
+          if (kpi) {
+            if (axis.key === 'number_of_transactions' || axis.key === 'total_cost') {
+              kpi.isWeighted = false;
+              kpi.weighted_average = kpi.value;
+            } else {
+              if (kpi.volume) {
+                weightedAverage = (kpi.valueTimesVolume / kpi.volume);
+                kpi.weighted_average = Math.round(weightedAverage * 1000) / 1000;
```
Comment (jcbashdown) [inline app/common/collections/services.js]: is there a more readable to do this.

File: app/vendor/jquery-deparam.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
+/*
+ jQuery deparam is an extraction of the deparam method from Ben Alman's jQuery BBQ
```
Comment (jcbashdown) [inline app/vendor/jquery-deparam.js]: do we need this if this is in package.json?

File: app/support/stagecraft_stub/responses/transaction-data.json [hunk 4]
```diff
@@ -0,0 +1,616 @@
+{
```
Comment (jcbashdown) [inline app/support/stagecraft_stub/responses/transaction-data.json]: move to backdrop stub

File: app/client/accessibility.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+define(['jquery'],
```
Comment (jcbashdown) [inline app/client/accessibility.js]: test

File: app/server/views/services.js [hunk 6]
```diff
@@ -23,25 +36,42 @@ module.exports = BaseView.extend({
     ];
   },
 
-  getContent: function () {
+  getAggregateValues: function () {
+    var aggVals = this.filterCollection.getAggregateValues();
+
+    _.each(aggVals, function (kpi) {
+      if (kpi.weighted_average) {
```
Comment (jcbashdown) [inline app/server/views/services.js]: Is there a test for this?

**Comment by jabley** [comment]
Can we have new lines at the end of the new files please? Cleaner future diffs FTW.

File: app/server/templates/services.html [hunk 7]
```diff
@@ -1,37 +1,103 @@
...
+          <% if (filteredCount > 1) { %>
+          <% if (kpi.isWeighted) { %>
+            weighted average
+          <% } else { %>
+            total
+          <% } %> for
+          <% } %>
+          <a href="#filtered-list" class="js-sort-by" data-sort-by="<%= kpi.key %>">
+          <% if (kpi.allRowsHaveValues == true) { %>
+            <% if (filteredCount == 1) { %>
```
Comment (jabley) [inline app/server/templates/services.html]: `==` as opposed to `===`?

File: app/server/templates/services.html [hunk 8]
```diff
@@ -1,37 +1,103 @@
...
+              all <%= filteredCount %> services
+            <% } %>
+          <% } else { %>
+          <%= kpi.valueCount %> services out of <%= filteredCount %>
+          <% } %>
+          </a>
+        </div>
+      </div>
+    </section>
+    <% if (index == 2) { %>
```
Comment (jabley) [inline app/server/templates/services.html]: `==` and feels a little Magic Number-y.

File: app/client/accessibility.js (hunk 5 above)
Comment (jonnywyatt) [inline app/client/accessibility.js]: Done

File: app/client/views/table.js [hunk 9]
```diff
@@ -162,6 +208,33 @@ function (TableView, Modernizr) {
...
+
+    screenreaderAnnounceSortChange: function(sortField, sortDirection) {
+      var sortLabel = this.sortFields[sortField] && this.sortFields[sortField].label;
+      if (sortLabel) {
+        accessibility.updateLiveRegion('Transactions now sorted by ' + sortLabel + ' in ' + sortDirection + ' order');
+      }
+    },
+
+    updateUrlWithSort: function() {
+      if (Modernizr.history) {
```
Comment (jabley) [inline app/client/views/table.js]: *twitches* If we we using a cut-down version of modernizr, then maybe. All of it? :cry:

File: app/server/templates/services.html (hunk 8 above)
Comment (jonnywyatt) [inline app/server/templates/services.html]: that's to ensure 3 modules per row. I could add a code comment?

File: app/common/collections/services.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                  abbr: true
+                }
+              });
+            }
+          });
+        });
+
+        _.each(axes, function (axis) {
+          var key = (axis.format === 'percent' || axis.format.type === 'percent') ?
+              'percentages' : 'totals',
```
Comment (jabley) [inline app/common/collections/services.js]: I would be happy to see this in a function. I **like** small, well-named functions.

File: app/support/stagecraft_stub/responses/transaction-data.json (hunk 4 above)
Comment (jonnywyatt) [inline app/support/stagecraft_stub/responses/transaction-data.json]: Done

File: app/client/views/table.js (hunk 9 above)
Comment (jonnywyatt) [inline app/client/views/table.js]: The codebase seems to use Modernizr tests for CORS, touch, inline SVG, history.

We seem to be using a custom build of Modernizr with support for only those. The comment at the top of the file gives a URL to reproduce the custom build.

File: app/common/collections/services.js (hunk 2 above)
Comment (jonnywyatt) [inline app/common/collections/services.js]: done

**Pull request merged**